### PR TITLE
Persistence: MongoDB

### PR DESCRIPTION
**Pull request by DonatJR**

### CODE REVIEW

File: src/main/scala/de/htwg/se/sudoku/SudokuModule.scala [hunk 1]
```diff
@@ -47,5 +46,26 @@ class MicroSudokuModule extends AbstractModule with ScalaModule {
 
     bind[FileIOInterface].to[fileIoMicroImpl.FileIO]
   }
+}
+
+class MongoDBModule extends AbstractModule with ScalaModule {
+
+  val defaultSize: Int = 9
+  val defaultHostname: String = "localhost"
+  val defaultFilePort: Int = 27017
```
Comment (SailReal) [inline src/main/scala/de/htwg/se/sudoku/SudokuModule.scala]: Maybe rename `defaultFilePort` to `defaultMongoDBPort`?

File: src/main/scala/de/htwg/se/sudoku/model/fileIoComponent/fileIoMongoDBImpl/FileIO.scala [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+class FileIO @Inject()(@Named("MongoDBHost") host: String, @Named("MongoDBPort") port: Int) extends FileIOInterface {
+
+  private val mongoClient: MongoClient = MongoClient(s"mongodb://$host:$port")
+  private val db: MongoDatabase = mongoClient.getDatabase("sudoku-in-scala")
+  private val collection = db.getCollection("sudoku-in-scala-col")
+
+  override def load: Try[Option[GridInterface]] = {
+    val resF = collection.find().projection(Projections.excludeId()).toFuture()
+    val res = Await.result(resF, Duration.Inf)
```
Comment (SailReal) [inline src/main/scala/de/htwg/se/sudoku/model/fileIoComponent/fileIoMongoDBImpl/FileIO.scala]: Maybe rename `res` to `result`? I don't think `res` is very meaningful. The same with `resF`.

File: src/main/scala/de/htwg/se/sudoku/model/fileIoComponent/fileIoMongoDBImpl/FileIO.scala [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+    var gridOption: Option[GridInterface] = None
+
+    Try {
+      val json: JsValue = Json.parse(res.head.toJson())
+      val size = (json \ "grid" \ "size").get.toString.toInt
+      val injector = Guice.createInjector(new SudokuModule)
```
Comment (SailReal) [inline src/main/scala/de/htwg/se/sudoku/model/fileIoComponent/fileIoMongoDBImpl/FileIO.scala]: Why `SudokuModule` and not `MongoDBModule`?

File: src/main/scala/de/htwg/se/sudoku/model/fileIoComponent/fileIoMongoDBImpl/FileIO.scala [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+      }
+      gridOption
+    }
+  }
+
+  override def save(grid: GridInterface): Try[Unit] = {
+    Try {
+      Await.result(collection.drop().toFuture(), Duration.Inf)
+      val gameStateDoc = Document.apply(gridToJson(grid).toString())
+      val resF = collection.insertOne(gameStateDoc).toFuture()
```
Comment (SailReal) [inline src/main/scala/de/htwg/se/sudoku/model/fileIoComponent/fileIoMongoDBImpl/FileIO.scala]: Same as above with `resF`...

File: src/main/scala/de/htwg/se/sudoku/model/fileIoComponent/fileIoMongoDBImpl/FileIO.scala (hunk 3 above)
Comment (DonatJR) [reply src/main/scala/de/htwg/se/sudoku/model/fileIoComponent/fileIoMongoDBImpl/FileIO.scala]: `MongoDBModule` only works if a mongo db database is installed, didn't want to make it the default

Comment (SailReal) [reply src/main/scala/de/htwg/se/sudoku/model/fileIoComponent/fileIoMongoDBImpl/FileIO.scala]: The `src/main/scala/de/htwg/se/sudoku/model/fileIoComponent/fileIoMongoDBImpl/FileIO.scala ` is only used if [this](https://github.com/bb30/de.htwg.se.SudokuInScala/pull/14/files#diff-2f2697619d766493c74a61548e98ce43R26) line is inserted again or? If so we can use here the `MongoDBModule` and when we wan't to use it, we have to change only the other line not both...

Comment (DonatJR) [reply src/main/scala/de/htwg/se/sudoku/model/fileIoComponent/fileIoMongoDBImpl/FileIO.scala]: ah sorry, didn't catch that we weren't talking about the line you linked to

**Pull request merged**